src/System.Net.Http.WinHttpHandler/tests/UnitTests/WinHttpHandlerTest.cs

```diff
@@ -846,6 +846,20 @@ public void SendAsync_NonSecureRequestToServer_ExpectNoClientCertContext()
                 async () => { await client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cts.Token); });
         }
 
+        [Fact]
+        public async Task SendAsync_WinHttpOpenReturnsError_ExpectHttpRequestException()
+        {
+            var handler = new WinHttpHandler();
+            var client = new HttpClient(handler);
+            var request = new HttpRequestMessage(HttpMethod.Get, FakeServerEndpoint);
+
+            TestControl.Fail.WinHttpOpen = true;
+
+            Exception ex = await Assert.ThrowsAsync<HttpRequestException>(
+                async () => { await client.SendAsync(request); });
```

Comment:
And actually this code doesn't compile:

`Exception ex = await Assert.ThrowsAsync(() => client.SendAsync(request));`

I get error:

WinHttpHandlerTest.cs(858,41): error CS0411: The type arguments for method 'Assert.ThrowsAsync<T>(Func<Task>)' cannot be inferred from the usage. Try specifying the type arguments explicitly.